src/lib/caches/Items.js

```diff
@@ -75,14 +75,20 @@ module.exports = class ItemsCache {
       hydrate();
       base = ItemsCache.#cache.getKey(`${language}-${key}`);
     }
+    // Allow nested keys separated by periods
+    const bys = by.split('.');
     let filtered = base;
     if (term && max !== 1) {
       filtered = base
-        .filter((item) => item && item[by])
+        .filter((item) => item && this.#parseNestedKey(item, bys))
         .map((item) => {
-          if (!(item && item[by])) return undefined;
-          if (item[by].toLowerCase().includes(term.toLowerCase())) {
-            return item;
+          const values = this.#parseNestedKey(item, bys);
+          if (!(item && values)) return undefined;
+          // parseNestedKey returns an array, so check all values in that array
+          for (const entry of values) {
+            if (entry.toString().toLowerCase().includes(term.toString().toLowerCase())) {
+              return item;
+            }
           }
```

Comment:
```js
          return values.find((entry) => entry.toString().toLowerCase().includes(term.toString().toLowerCase());
```

array.find returns the first entry for which the return value is true from the filtering function. if we need multiple, we could return `values.filter` with the same filter function

and if you want to enforce undefined instead of null, you can ` || undefined` at the end


---

rereading this, i think the intention was to return on this only if it's got a value, so instead...

```suggestion
          const found = values.find((entry) => entry.toString().toLowerCase().includes(term.toString().toLowerCase());
          if (found) return found;
```

Reply:
Adopted find within an if. Returning item instead of return value since return value would be the matched value not the whole item.
```javascript
          if (values.find((entry) => entry.toString().toLowerCase().includes(term.toString().toLowerCase()))) {
            return item;
          }
```